Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial:IEquatable<Polynomial>
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length>100)
+            {
+                throw new Exception("Max power must be less equal 100");
+            }
+            _coefficients = coefficients;
+        }
+
+        public bool Equals(Polynomial? other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (!(other is Polynomial || this is Polynomial) )
+            {
+                return false;
+            }
+
+            if (this._coefficients.Length != other._coefficients.Length)
+            {
+                return false;
+            }
+
+            if (this == other)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool FloatsAreEqualEnough(float left, float right)
+        {
+            const double tolerance = 0.000000001;
+            if (!(Math.Abs(left - right) < tolerance))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static Polynomial operator +(Polynomial? left, Polynomial? right)
+        {
+            if (left is null || !(right is null))
+            {
+                return right;
```

Comment:
When you do some math operations, you woudn't expect that operation to returb exactly the same object. Please make there a new object, which is an exact copy of this one (as an option, consider implementing ICloneable interface)